bootloader/grub.go

```diff
@@ -68,13 +69,36 @@ func (g *grub) InstallBootConfig(gadgetDir string, opts *Options) (bool, error)
 	if opts != nil && opts.Recovery {
 		recoveryGrubCfg := filepath.Join(gadgetDir, g.Name()+"-recovery.conf")
 		systemFile := filepath.Join(g.rootdir, "/EFI/ubuntu/grub.cfg")
-		return genericInstallBootConfig(recoveryGrubCfg, systemFile)
+		ok, err := genericInstallBootConfig(recoveryGrubCfg, systemFile)
+		if !ok {
+			return false, nil
+		}
+		if err != nil {
+			return true, err
+		}
+		return true, g.setRecoveryKernel(opts.RecoverySystem, opts.RecoveryKernel)
 	}
 	gadgetFile := filepath.Join(gadgetDir, g.Name()+".conf")
 	systemFile := filepath.Join(g.rootdir, "/boot/grub/grub.cfg")
 	return genericInstallBootConfig(gadgetFile, systemFile)
 }
 
+func (g *grub) setRecoveryKernel(recoverySystem, kernel string) error {
```

Comment:
this seems splitting responsabilites along the wrong lines, it the first time in bootloader that we use/know directly snap_* boot env vars (or about systems), we need to divide this with boot differently. OTOH we don't know how to do this for other bootloaders yet.  Maybe a method only on grub  SetRecoverySystemEnv(recoverySystemDir, values map[string]string), and then using an interface in boot to detect if it is applicable.

Reply:
To be clear, this means not changing InstallBootConfig itself but calling that one separately and explicitly in boot.